core/ldap_api.php

```diff
@@ -379,7 +379,11 @@ function ldap_authenticate_by_username( $p_username, $p_password ) {
 
 		if( false !== $t_user_id ) {
 
-			$t_fields_to_update = array('password' => md5( $p_password ));
+			$t_fields_to_update = array();
+			# Do we need to wipe a previously cached password?
+			if( user_get_field( $t_user_id, 'password' )[0] != '!' ) {
```

Comment:
Function array dereferencing is not supported in PHP 5.3, which is our minimal supported version.